umbraco/Umbraco-CMS
# Fix mandatory RTE validation

### Description
This fixes a bug, where the mandatory check did not work on RTEs, after the markdown have been wrapped in json.

To fix it, a new custom RichTextRequiredValidator have been introduced. It inherits the original RequiredValidator and passes the markdown to that.

### Test
- Setup a doc type with a RTE and make it mandatory.
- Check that you get an correct validation message when publishing an empty RTE.
 
<img width="1151" alt="image" src="https://github.com/user-attachments/assets/bc52e48d-9e8f-4021-a7f4-4fa2c2d0f33a">


### Notes for when merged to v14
- In v14, we do not need to inject ILocalizedTextService into RichTextRequiredValidator. So we should introduce an overload of the constructor that do not use this, and obsolete the one in this PR
